src/System.Private.CoreLib/shared/Internal/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -387,5 +387,48 @@ public static IntPtr ByteOffset<T>(ref T origin, ref T target)
         {
             throw new PlatformNotSupportedException();
         }
+
+        /// <summary>
+        /// Returns a by-ref that is a null reference to a type <typeparamref name="T"/>.
+        /// Note this is the address of the by-ref not what the ref points to.
+        /// </summary>
+        [Intrinsic]
+        [NonVersionable]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static ref T NullRef<T>()
```

Comment:
Nit: You can make these two methods public

Reply:
Done